File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/user/UserLiteWithEventDTO.java
package es.jocasolo.competitiveeventsapi.dto.user;

import java.io.Serializable;
import java.util.Date;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.dto.image.ImageDTO;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserPrivilegeType;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserStatusType;
import lombok.Getter;
import lombok.Setter;

public class UserLiteWithEventDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private String id;

	@Getter
	@Setter
	private String name;

	@Getter
	@Setter
	private String surname;

	@Getter
	@Setter
	private ImageDTO avatar;
	
	@Getter
	@Setter
	private Date incorporationDate;
	
	@Getter
	@Setter
	private Date lastStatusDate;
	
	@Getter
	@Setter
	private EventUserPrivilegeType privilege;
	
	@Getter
	@Setter
	private EventUserStatusType status;

	@Override
	public String toString() {
		return "UserLiteWithEventDTO [id=" + id + "]";
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/user/UserLiteWithEventDTO.kt
package es.jocasolo.competitiveeventsapp.dto.user

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.HistoryItemDTO
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserPrivilegeType
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserStatusType
import java.time.LocalDateTime
import java.util.*

class UserLiteWithEventDTO(
    @SerializedName("id") var id: String? = null,
    @SerializedName("name") var name: String? = null,
    @SerializedName("surname") var surname: String? = null,
    @SerializedName("description") var description: String? = null,
    @SerializedName("birthDate") var birthDate: Date? = null,
    @SerializedName("registerDate") var registerDate: Date? = null,
    @SerializedName("email") var email: String? = null,
    @SerializedName("avatar") var avatar: ImageDTO? = null,
    @SerializedName("incorporationDate") var incorporationDate: Date?,
    @SerializedName("lastStatusDate") var lastStatusDate: Date?,
    @SerializedName("privilege") var privilege: EventUserPrivilegeType?,
    @SerializedName("status") var status: EventUserStatusType?
) : HistoryItemDTO() {

}